src/nft/components/bag/ButtonStates.tsx

```diff
@@ -0,0 +1,148 @@
+import { Trans } from '@lingui/macro'
+import { PriceImpact } from 'nft/hooks/usePriceImpact'
+import { ReactNode } from 'react'
+import { DefaultTheme } from 'styled-components/macro'
+
+export enum BuyButtonStates {
+  WALLET_NOT_CONNECTED,
+  NOT_SUPPORTED_CHAIN,
+  INSUFFICIENT_BALANCE,
+  ERROR,
+  IN_WALLET_CONFIRMATION,
+  PROCESSING_TRANSACTION,
+  FETCHING_TOKEN_ROUTE,
+  INVALID_TOKEN_ROUTE,
+  NO_TOKEN_ROUTE_FOUND,
+  LOADING_ALLOWANCE,
+  IN_WALLET_ALLOWANCE_APPROVAL,
+  PROCESSING_APPROVAL,
+  REQUIRE_APPROVAL,
+  CONFIRM_UPDATED_PRICE,
+  PRICE_IMPACT_HIGH,
+  PAY,
+}
+
+export interface BuyButtonState {
+  handleClick: (() => void) | (() => Promise<void>)
+  buttonText: ReactNode
+  disabled: boolean
+  warningText: ReactNode | undefined
+  warningTextColor: string
+  helperText: ReactNode | undefined
+  helperTextColor: string
+  buttonColor: string
+  buttonTextColor: string
+}
+
+export function getBuyButtonStateData(
```

Comment:
Can you add unit tests for this fn?

Reply:
unit tests don't really make sense for this function, what would it be defending against. If a future dev adds to the `BuyButtonStates` enum and doesn't add an appropriate case eslint will throw an error and vice versa

Reply:
as you can see adding a state throws an eslint error

![Screen Shot 2023-05-23 at 1 08 04 PM](https://github.com/Uniswap/interface/assets/9094792/18037061-2da7-4ab7-852a-fa55c5b63533)

Reply:
You can pass in the various different `BuyButtonStates` and expect the corresponding `BuyButtonStateData`. This will help prevent future devs from unknowingly making changes to the `BuyButtonStateData` possible leading to a `BuyButton` regression